lib/rbUtil.js

```diff
@@ -275,13 +275,19 @@ rbUtil.parseETag = function(etag) {
  * Copies forwarded headers from restbase to request.
  * If the same header was already set it takes precedence over
  * the forwarded header.
+ *
+ * @param restbase a restbase instance to take the forwarded headers from
+ * @param req a request where to copy the headers
+ * @param headers array of header names to copy
  */
-rbUtil.copyForwardedCookies = function(restbase, req) {
-    if (restbase._rootReq
-            && restbase._forwardedCookies
-            && !req.headers.cookie) {
+rbUtil.copyForwardedHeaders = function(restbase, req, headers) {
+    if (restbase._rootReq && restbase._forwardedHeaders) {
         req.headers = req.headers || {};
-        req.headers.cookie = restbase._forwardedCookies;
+        headers.forEach(function(header) {
```

Comment:
Does it make sense here to allow `headers` not to be passed in, in which case all of the `restbase._forwardedHeaders` would be used?